packages/flutter_tools/test/integration.shard/command_output_test.dart

```diff
@@ -128,4 +130,20 @@ void main() {
     // Only printed by verbose tool.
     expect(result.stdout, isNot(contains('exiting with code 0')));
   });
+
+  test('flutter --version --machine outputs JSON with flutterRoot', () async {
+    final String flutterBin = globals.fs.path.join(getFlutterRoot(), 'bin', 'flutter');
+    final ProcessResult result = await const LocalProcessManager().run(<String>[
+      flutterBin,
+      '--version',
+      '--machine',
+    ]);
+
+    final Map<String, Object> versionInfo = json.decode(result.stdout
+      .toString()
+      .replaceAll('Building flutter tool...', '')
```

Comment:
This has been around forever, but it'd be really nice if we could suppress the non-JSON text like this when `--machine` is in the args 😄

Reply:
It should be safe to write this to stderr too, right? I was looking at this a while back but that change ended being hard to land due to integration tests asserting stderr was empty

Reply:
Hmmm, not sure.. I think there are probably many places that assume stderr is for errors. I think PowerShell may also (or at least used to?) print stderr in scary colors.. some common tool like Git was always causing my grief because it wrote some status messages to stderr and made it look like it'd crashed 😞

Reply:
ack